### PR TITLE
Filler lite

### PR DESCRIPTION
**Pull request by NourAlharithi**
It's filler time

### CODE REVIEW

File: src/bots/fillerLite.ts [hunk 1]
```diff
@@ -0,0 +1,1274 @@
...
+	}
+
+	public async startIntervalLoop(_intervalMs: number) {
+		const intervalId = setInterval(
+			this.tryFill.bind(this),
+			this.pollingIntervalMs
+		);
+		this.intervalIds.push(intervalId);
+
+		this.eventSubscriber.eventEmitter.on(
```
Comment (crispheaney) [inline src/bots/fillerLite.ts:389]: could prob get rid of this @wphan ?

File: src/bots/fillerLite.ts [hunk 2]
```diff
@@ -0,0 +1,1274 @@
...
+
+	/**
+	 * Iterates through a tx's logs and handles it appropriately 3e.g. throttling users, updating metrics, etc.)
+	 *
+	 * @param nodesFilled nodes that we sent a transaction to fill
+	 * @param logs logs from tx.meta.logMessages or this.clearingHouse.program._events._eventParser.parseLogs
+	 *
+	 * @returns number of nodes successfully filled
+	 */
+	protected async handleTransactionLogs(
```
Comment (wphan) [inline src/bots/fillerLite.ts:492]: a lot of this is same as the parent implementation, perhaps pass in a function to retrieve the `UserAccount`, then filler lite can use `orderSubscriber`, and parent can keep using `UserMap`?

File: src/bots/fillerLite.ts [hunk 3]
```diff
@@ -0,0 +1,1274 @@
...
+			if (this.fillTxSinceBurstCU > TX_COUNT_COOLDOWN_ON_BURST) {
+				this.useBurstCULimit = false;
+			}
+			this.fillTxSinceBurstCU += 1;
+		}
+
+		return successCount;
+	}
+
+	protected async sendFillTx(
```
Comment (wphan) [inline src/bots/fillerLite.ts:693]: looks pretty much identical to parent implementation

File: src/bots/fillerLite.ts [hunk 4]
```diff
@@ -0,0 +1,1274 @@
...
+			if (!isVariant(marketType, 'perp')) {
+				throw new Error('expected perp market type');
+			}
+
+			ixs.push(
+				await this.driftClient.getFillPerpOrderIx(
+					await getUserAccountPublicKey(
+						this.driftClient.program.programId,
+						takerUser.authority
+					),
```
Comment (wphan) [inline src/bots/fillerLite.ts:907]: dont think this PDA is correct, since it is assuming subaccountId 0 for the taker.

Comment (wphan) [reply src/bots/fillerLite.ts:907]: oh you can get that from `takerUser.subaccountId`

File: src/bots/fillerLite.ts [hunk 5]
```diff
@@ -0,0 +1,1274 @@
...
+					this.removeFillingNodes(nodesSent);
+				});
+		}
+	}
+
+	/**
+	 * It's difficult to estimate CU cost of multi maker ix, so we'll just send it in its own transaction
+	 * @param node node with multiple makers
+	 */
+	protected async tryFillMultiMakerPerpNodes(nodeToFill: NodeToFill) {
```
Comment (wphan) [inline src/bots/fillerLite.ts:877]: also looks pretty similar to parent implementation, consider merging em

File: src/bots/fillerLite.ts [hunk 6]
```diff
@@ -0,0 +1,1274 @@
...
+		} catch (e) {
+			logger.error(
+				`Error filling multi maker perp node (fillTxId: ${fillTxId}): ${
+					e.stack ? e.stack : e.message
+				}`
+			);
+		}
+	}
+
+	async tryBulkFillPerpNodes(nodesToFill: Array<NodeToFill>): Promise<number> {
```
Comment (wphan) [inline src/bots/fillerLite.ts:931]: consider using parent's

File: src/bots/fillerLite.ts [hunk 7]
```diff
@@ -0,0 +1,1274 @@
...
+		if (this.revertOnFailure) {
+			ixs.push(await this.driftClient.getRevertFillIx());
+		}
+
+		this.sendFillTx(fillTxId, nodesSent, ixs);
+
+		return nodesSent.length;
+	}
+
+	protected filterPerpNodesForMarket(
```
Comment (wphan) [inline src/bots/fillerLite.ts:1093]: can you parents here

File: src/bots/fillerLite.ts [hunk 8]
```diff
@@ -0,0 +1,1274 @@
...
+			return this.filterTriggerableNodes(node);
+		});
+
+		return {
+			filteredFillableNodes,
+			filteredTriggerableNodes,
+		};
+	}
+
+	protected async executeTriggerablePerpNodesForMarket(
```
Comment (wphan) [inline src/bots/fillerLite.ts:1126]: consider merging with parent

File: src/bots/fillerLite.ts [hunk 9]
```diff
@@ -0,0 +1,1274 @@
...
+		this.attemptedTriggersCounter.add(
+			triggerableNodes.length,
+			metricAttrFromUserAccount(
+				user.userAccountPublicKey,
+				user.getUserAccount()
+			)
+		);
+	}
+
+	protected async tryFill() {
```
Comment (wphan) [inline src/bots/fillerLite.ts:1190]: looks like this guy can be largely using the parent too! just need to refactor to allow the option to use `ordersubscriber`

File: src/index.ts [hunk 10]
```diff
@@ -234,20 +245,32 @@ const runBot = async () => {
 		};
 	}
 
+	const { perpMarketIndexes, spotMarketIndexes, oracleInfos } =
+		getMarketsAndOraclesForSubscription(config.global.driftEnv);
```
[the remark this answers is not in the document]
Comment (NourAlharithi) [reply src/index.ts:249]: sounds good i'll put a conditional there

**Pull request merged**